crates/net/eth-wire/src/p2pstream.rs

```diff
@@ -123,7 +123,12 @@ where
         let their_hello = match P2PMessage::decode(&mut &first_message_bytes[..]) {
             Ok(P2PMessage::Hello(hello)) => Ok(hello),
             Ok(P2PMessage::Disconnect(reason)) => {
-                tracing::debug!("Disconnected by peer during handshake: {}", reason);
+                if matches!(reason, DisconnectReason::TooManyPeers) {
+                    // Too many peers is a very common disconnect reason that spams the DEBUG logs
+                    tracing::trace!("Disconnected by peer during handshake: {}", reason)
+                } else {
+                    tracing::debug!("Disconnected by peer during handshake: {}", reason)
+                }
```

Comment:
nit
```suggestion
                let level = match reason {
                    // Too many peers is a very common disconnect reason that spams the DEBUG logs
                    DisconnectReason::TooManyPeers => Level::Trace,
                    _ => Level::Debug,
                };
                event!(level, %reason, "Disconnected by peer during handshake");
```

Reply:
`attempt to use a non-constant value in a constant` 🥲  because macro has a static callsite definition